packages/lesswrong/components/form-components/ImageUpload.tsx

```diff
@@ -139,47 +146,67 @@ class ImageUpload extends Component<any,any> {
         }
       },
       ...cloudinaryArgs
-    }, this.setImageInfo);
+    }, setImageInfo);
   }
   
-  removeImg = () => {
-    this.props.clearField();
-    this.setState({imageId: null});
+  const chooseDefaultImg = (newImageId) => {
+    setImageId(newImageId)
+    updateCurrentValues({[name]: newImageId})
   }
   
-  render(){
-    const { classes, name, label } = this.props;
-    const formPreviewSize = formPreviewSizeByImageType[name]
-    if (!formPreviewSize) throw new Error("Unsupported image upload type")
-    
-    return (
-      <div className={classes.root}>
-        <Helmet>
-          <script src="https://upload-widget.cloudinary.com/global/all.js" type="text/javascript"/>
-          <script src='//ajax.googleapis.com/ajax/libs/jquery/1.11.1/jquery.min.js'/>
-        </Helmet>
-        {this.state.imageId &&
-          <Components.CloudinaryImage
-            publicId={this.state.imageId}
-            {...formPreviewSize}
-          /> }
-        <Button
-          onClick={this.uploadWidget}
-          className={classNames("image-upload-button", classes.button)}
-        >
-          <ImageIcon className={classes.imageIcon}/>
-          {this.state.imageId ? `Replace ${label}` : `Upload ${label}`}
-        </Button>
-        {this.state.imageId && <Button
-          className={classes.removeButton}
-          title="Remove"
-          onClick={this.removeImg}
-        >
-          Remove {label}
-        </Button>}
-      </div>
-    );
+  const removeImg = () => {
+    clearField()
+    setImageId(null)
   }
+  
+  const { openDialog } = useDialog()
+  const [imageId, setImageId] = useState(() => {
+    if (document && document[name]) {
+      return document[name];
+    }
+    return ''
+  })
+  
+  const formPreviewSize = formPreviewSizeByImageType[name]
+  if (!formPreviewSize) throw new Error("Unsupported image upload type")
+  
+  return (
+    <div className={classes.root}>
+      <Helmet>
+        <script src="https://upload-widget.cloudinary.com/global/all.js" type="text/javascript"/>
+        <script src='//ajax.googleapis.com/ajax/libs/jquery/1.11.1/jquery.min.js'/>
```

Comment:
Do we still need jquery? The cloudinary documentation seems to imply that only the old version needed it

Reply:
Good catch - image upload seems to work without it so I removed it